lib/serializers/team_serializer.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require_relative '../models/team'
+
+class TeamSerializer
+  class << self
+    def from_json(json, folders_json: [])
+      json = JSON.parse(json, symbolize_names: true)
+      data = json[:data] || json
+      attributes = data[:attributes]
+      folder_ids = data[:relationships][:folders][:data].map { |folder| folder[:id] }
+      folders = folders_json.map do |folder|
+        FolderSerializer.from_json(folder.to_json) if folder_ids.include?(folder[:id])
+      end.compact
+      Team.new(name: attributes[:name],
+               folders: folders,
+               id: data[:id])
+    end
+
+    def render_list(team)
```

Comment:
doesn't #render_list belong to team model ?

Reply:
Hmm it does. But in the model, I simply delegate to this method. This way, you can easily call the method on your instance and the logic is inside the serializer. Do you think that's an unecessary delegation?

Reply:
Method definition on model is here: https://github.com/puzzle/ccli/pull/11/files#diff-ac0ed614502ed7cfc515046f59dc52e1R17

Reply:
I just think this has really nothing to do with the serializer, does it? the serializer is responsible for converting json to data and versa. whereas #render_list renders a list of folders.
In my opinion, this belongs more to the model in your case. Probably it would belong to a view layer. maybe introduce for example presenters and move it over there ?

Reply:
from wikipedia: The presenter acts upon the model and the view. It retrieves data from repositories (the model), and formats it for display in the view.